packages/engine-core/src/library/LibraryEngine.ts

```diff
@@ -441,11 +441,7 @@ You may have to run ${chalk.greenBright('prisma generate')} for your changes to
     await this.start()
 
     this.lastQuery = JSON.stringify(request)
-    this.executingQueryPromise = this.engine!.query(
-      this.lastQuery,
-      JSON.stringify(headers), // TODO these aren't headers on the engine side
-      headers.transactionId,
-    )
+    this.executingQueryPromise = this.engine!.query(this.lastQuery, JSON.stringify(headers), headers.transactionId)
```

Comment:
I think the comment that those are not _really_ headers was helpful. The `TODO` can go though of course.

Reply:
```suggestion
    // Note that `headers` aren't headers on the engine side
    this.executingQueryPromise = this.engine!.query(this.lastQuery, JSON.stringify(headers), headers.transactionId)
```